src/richamster/common.rs

```diff
@@ -0,0 +1,101 @@
+use hmac_sha256::HMAC;
+use reqwest::RequestBuilder;
+use secrecy::{ExposeSecret, Secret};
+
+const HEADER_API_KEY: &str = "Api-Key";
+const HEADER_SIGNATURE: &str = "Signature";
+const HEADER_AUTH: &str = "Authorization";
+const JWT: &str = "JWT";
+
+#[derive(Debug, Clone)]
+pub struct JwtToken(pub Secret<String>);
+
+impl JwtToken {
+    fn value(&self) -> &str {
+        self.0.expose_secret()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ApiKey(pub Secret<String>);
+
+impl ApiKey {
+    fn value(&self) -> &str {
+        self.0.expose_secret()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct SecretKey(pub Secret<String>);
+
+impl SecretKey {
+    fn value(&self) -> &str {
+        self.0.expose_secret()
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum AuthState {
+    Unauthorized,
+    JwtTokenAuth(JwtToken),
+    ApiSecretKeyAuth(ApiKey, SecretKey),
+    JwtTokenWithApiSecretKeyAuth(JwtToken, ApiKey, SecretKey),
+}
```

Comment:
I saw that you are using `AuthState::Unauthorized` as default state in some places (e.g `User`). If I understand the code correctly, you can derive `Default` implementation here and set `#[default]` for `Unauthorised`.

Reply:
Done